kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/PatternViewModel.java

```diff
@@ -83,6 +111,32 @@ public PatternViewModel() {
                     .addProperty(OTHER_NAME_DESCRIPTION_NAME_TEXT, "")
                     // Ordered collection of Fields
                     .addProperty(FIELDS_COLLECTION, new ArrayList<PatternField>())
+                    .addProperty(IS_INVALID, true)
+                    .addValidator(IS_INVALID, "Is Invalid", (ValidationResult vr, ViewModel viewModel) -> {
+                        ObjectProperty<EntityFacade> purposeEntity = viewModel.getProperty(PURPOSE_ENTITY);
+                        ObjectProperty<EntityFacade> meaningEntity = viewModel.getProperty(MEANING_ENTITY);
+                        ObjectProperty<DescrName> fqnProperty = viewModel.getProperty(FQN_DESCRIPTION_NAME);
+                        ObservableList<PatternField> fieldsProperty = viewModel.getObservableList(FIELDS_COLLECTION);
+                        if (purposeEntity.isNull().get()) {
+                            vr.error("A purpose is required for a Pattern.  Please add a purpose.");
+                            viewModel.setPropertyValue(IS_INVALID, true);
+                        }
+                        if (meaningEntity.isNull().get()) {
+                            vr.error("A meaning is required for a Pattern.  Please add a meaning.");
+                            viewModel.setPropertyValue(IS_INVALID, true);
+                        }
+                        if (purposeEntity.isNull().get() == false && meaningEntity.isNull().get() == false) {
```

Comment:
Would this be easier to at the end check for `vr.getMessage().size() == 0`? 
```java
   if (vr.getMessage().size() == 0) {
       viewModel.setPropertyValue(IS_INVALID, false);
   }
```
**Tip:** Also boolean logic as short hand the `== false` you should use `!`.

Reply:
updated